src/main/java/ca/bc/gov/hlth/pbfdataloader/service/SFTPService.java

```diff
@@ -37,20 +37,17 @@ public void removeFile(String fileName) {
 	}
 
 	public File getFile(String fileName) {
-		File tempFile = null;
 		try (SSHClient sshClient = setupSshj(); SFTPClient sftpClient = sshClient.newSFTPClient()) {
-	    
-		    int separatorIndex = StringUtils.lastIndexOf(fileName, ".");
-		    String prefix = StringUtils.substring(fileName, 0, separatorIndex);
-		    String suffix = StringUtils.substringAfter(fileName, separatorIndex);
+			// All files need to be downloaded to a file to create a temp file and delete it later
+		    File tempFile = File.createTempFile(FilenameUtils.getBaseName(fileName), FilenameUtils.getExtension(fileName));
```

Comment:
Might be missing period before file suffix, file created would be named without it e.g. MSP_TPCPY_VW133463924793132550464179578203497043227csv. Note, unit tests add the period
File.createTempFile(FilenameUtils.getBaseName(inputFile.getName()), "." + FilenameUtils.getExtension(inputFile.getName()))
 which seems to work.